### DISCUSSION TITLE
Query filters not persisted when navigating back

### DISCUSSION THREAD

**Discussion by joostvanhoof** [livewire/livewire#3767] [7. Bugs (archived)]
It looks like the query string behaviour changed after the `2.6` update. I know there have been updates to the formatting but it seems that there has been a change to how it works when navigating back using the browser's back button.

Have look at [www.restory.nl/shop-kleding](https://restory.nl/shop-kleding) (it's in Dutch but no need to be able to speak the impossible language for this one). This is currently using the latest Livewire `2.6.3`.

**✅ This works as expected**
- In the filters, click on 'Meisje'
- Click on any product
- Use your browser's back button to go back to the search page
- The filter is still applied

**🛑 This doesn't**
- In the filters, click on 'Meisje'
- Click on any product
- Use your browser's back button to go back to the search page (same as scenario above, up to here it works as expected)
- Click on _any other_ filter
- Click on any product
- Use your browser's back button to go back to the search page
- Only the first filter ('Meisje') is applied, but not the second filter that you clicked

The non-working behaviour keeps going, so when you keep adding filters and go back, you'll see that the filters you have selected previously are being applied.

The only weird thing is that everything works fine locally, just not on production...

My only guess is that this could be related to the bug in routing and the URL behaviour that was present in `2.6.0` and `2.6.1` and was fixed in `2.6.2`.

**Comment by joshhanley** [comment]
@joostvanhoof other than an international site being slow from here in Australia 😂, it all seems to work fine with the correct query strings.

See below screen recording, let me know your thoughts 🙂

![recording (38)](https://user-images.githubusercontent.com/882837/133575623-d20630b0-9ae2-4844-b333-5edc4bf18d17.gif)

**Reply by joostvanhoof** [reply]
Yep that works as expected. Can you also try on Safari? I'm seeing different behaviour between these browsers.

**Reply by joostvanhoof** [reply]
@joshhanley I just downgraded to `2.5.5` (also on production) and the issue is gone. Happy to give you access to the repo to test if you want.

**Reply by joshhanley** [reply]
@joostvanhoof sorry didn't get a chance to play with it on safari while it was still live. Do you have a dev/staging server already setup that you could put the latest Livewire version on so I could browse to it in safari and test this?

**Reply by joostvanhoof** [reply]
@joshhanley I still have this bug on `2.7.0`. I'm one step further though: I can now reproduce the bug locally. I'll add you to the repo so that you can test. There is a `livewire-bug` branch with a `readme` and `.env-example`.

**Reply by joshhanley** [reply]
@joostvanhoof thanks! I will have a look!

**Reply by joshhanley** [reply]
@joostvanhoof once I have the app installed, what steps do I need to follow to replicate it or are they the same as above?

**Reply by joshhanley** [reply]
@joostvanhoof I followed the original steps on that local `livewire-bug` branch and it all seemed ok to me

**Reply by joostvanhoof** [reply]
@joshhanley first of all: congrats on officially joining the Livewire team! The bug is still there, indeed when you follow the steps as described in this issue. It does take some back and forth sometimes, it doesn't seem to be very consistent. So click some filters, go to a product page, go back, add some more filters, and then repeat this behaviour and at some point *some* of the selected filters will not be selected anymore. There are customers reporting this behaviour as well.

Because there was another bug (unrelated and now solved), this bug is currently live on production so you can see what happens there. It currently looks like the bug doesn't happen locally now, but it does on production.

Check out the last 5 seconds in the attached screen recording. I go back using the browser's back button (sorry this is not in the screen recording) and you can see that the filters flash being selected but are then deselected. Low quality video as Github accepts 10MB max, but it should clearly show what's happening.

https://user-images.githubusercontent.com/5268113/138428676-102b9e81-3630-4208-9388-aed403429bb4.mov

**Reply by joostvanhoof** [reply]
I have just downgrade Livewire to `2.5.5` on production and now the bug is gone again. So my suspicion is that it has to do with the updated `$queryString` formatting/behaviour that was introduced in `2.6`

**Reply by joshhanley** [reply]
@joostvanhoof thanks! 🙂

Ah sorry, keep missing each other when it's in production 😕 

I wonder why it is happening in production but we can't replicate it locally. Makes it a lot harder to try and troubleshoot. One thing that can be checked is checking the history in your browser, and seeing if the all the querystring parameters are being persisted to the browser history correctly as that might help narrow down the problem.
